README.md

```diff
@@ -1,68 +1,302 @@
+
+
+## Pull request labels
+
+The bot uses the following custom GitHub labels to mark the current pull
+request state:
+
+* `M-waiting-approvals`: The PR approval rules (see "Voting and PR
+  approvals" below) do not allow the bot to start merging regardless of
+  how old the PR gets. Compare with `M-waiting-objections`. The bot
+  removes this label when it notices that it is obsolete. TODO:
+  Implement.
+* `M-waiting-objections`: The PR approval rules (see "Voting and PR
+  approvals" below) allow the bot to start merging when the PR gets
+  older. The bot will eventually start merging this PR if no objections
+  are received while the bot is waiting. The wait time is determined by
+  the number of votes already received. The bot removes this label when
+  it notices that it is obsolete. TODO: Implement.
+* `M-waiting-staging-checks`: The bot determined that the PR is eligible
+  for merging, successfully completed the initial merging steps that are
+  supposed to trigger CI tests, and is now waiting for the CI test
+  results. The bot removes this label when it notices that all tests
+  have completed.
+* `M-passed-staging-checks`: Similar to the GitHub "green check" mark
+  for the staging branch commit (but ignores failures of optional
+  checks). Applied only if `config::skip_merge` option is on. The bot
+  removes this label when either PR was successfully merged or staging
+  results are no longer fresh/applicable.
+* `M-failed-staging-checks`: Essentially duplicates GitHub "red x" mark
+  for the _staging commit_. The bot removes this label when it notices
+  that the failed checks are no longer fresh/applicable.
+* `M-failed-description`: The PR title and/or description is invalid
+  (see below for PR commit message rules). The bot removes this label
+  when it revisits the PR and notices that the commit message components
+  were fixed.
+* `M-failed-other`: A fatal PR-specific error other than the staging
+  branch test failure (the latter is marked with
+  `M-failed-staging-checks`). It is probably necessary to
+  consult CI logs to determine what happened. The bot does not attempt
+  to merge this PR again until the PR branch or PR target branch change.
+  When the bot notices that change, it removes this label.
+* `M-merged`: The PR was successfully merged (and probably closed).
+  The bot will not attempt to merge this PR again even if it is
+  reopened. The bot never removes this label.
+
+All labels, except `M-merged`, are ignored by the bot itself! Humans
+may find them useful when determining the current state of a PR.
+
+
+## Commit message
+
+The staging commit message and, hence, the target branch commit is
+formed by concatenating the PR title (with an appended PR number), and
+empty line, and the PR description.
+
+Neither the title nor the description are currently processed to convert
+GitHub markdown to plain text. However, both texts must conform to the
+72 characters/line limit. PRs violating this limit are labeled
```

Comment:
We check the commit message lines _after_ adding the PR number. This is probably the right thing to do, but it places more stringent requirements on the title. The length of the added ` (#NNN)` title suffix is 7. Let's add something like this after the `However, ...` sentence:

```
The automatically added ` #(NNN)` title suffix further reduces the maximum PR title length
to ~65 characters.
```

and s/violating this limit/violating these limits/.

Reply:
Done.